services/cirrus/cirrus.service.js

```diff
@@ -0,0 +1,83 @@
+'use strict'
+
+const Joi = require('@hapi/joi')
+const { isBuildStatus, renderBuildStatusBadge } = require('../build-status')
+const { BaseJsonService } = require('..')
+
+const schema = Joi.object({
+  status: Joi.alternatives()
+    .try(isBuildStatus, Joi.equal('unknown'))
+    .required(),
+}).required()
+const queryParamSchema = Joi.object({
+  task: Joi.string(),
+  script: Joi.string(),
+}).required()
+
+module.exports = class Cirrus extends BaseJsonService {
+  static get category() {
+    return 'build'
+  }
+
+  static get route() {
+    return {
+      base: 'cirrus',
+      pattern: 'github/:user/:repo/:branch*',
+      queryParamSchema,
+    }
+  }
+
+  static get examples() {
+    return [
+      {
+        title: 'Cirrus CI - Base Branch Build Status',
+        namedParams: { user: 'flutter', repo: 'flutter' },
+        pattern: '/github/:user/:repo',
+        queryParams: { task: 'analyze', script: 'test' },
+        staticPreview: this.render({ subject: 'Ci', status: 'passing' }),
+      },
+      {
+        title: 'Cirrus CI - Specific Branch Build Status',
+        pattern: '/github/:user/:repo/:branch',
+        namedParams: { user: 'flutter', repo: 'flutter', branch: 'master' },
+        queryParams: { task: 'analyze', script: 'test' },
+        staticPreview: this.render({ subject: 'Ci', status: 'passing' }),
+      },
+      {
+        title: 'Cirrus CI - Specific Task Build Status',
+        pattern: '/github/:user/:repo',
+        queryParams: { task: 'analyze' },
+        namedParams: { user: 'flutter', repo: 'flutter' },
+        staticPreview: this.render({ subject: 'analyze', status: 'passing' }),
+      },
+      {
+        title: 'Cirrus CI - Task and Script Build Status',
+        pattern: '/github/:user/:repo',
+        queryParams: { task: 'analyze', script: 'test' },
+        namedParams: {
+          user: 'flutter',
+          repo: 'flutter',
+        },
+        staticPreview: this.render({ subject: 'test', status: 'passing' }),
+      },
+    ]
+  }
+
+  static get defaultBadgeData() {
+    return { label: 'build' }
+  }
+
+  static render({ subject, status }) {
+    return renderBuildStatusBadge({ label: subject, status })
+  }
+
+  async handle({ user, repo, branch }, { script, task }) {
+    const json = await this._requestJson({
+      schema,
+      url: `https://api.cirrus-ci.com/github/${user}/${repo}.json`,
+      options: { qs: { branch, script, task } },
+    })
+
+    return this.constructor.render({ status: json.status })
```

Comment:
@calebcartwright did you mean something like this to propagate either task name or a script name as a label?

```suggestion
    return this.constructor.render({ subject: task || script || 'build', status: json.status })
```

Reply:
@fkorotov - yes, generally speaking.

I wasn't sure if there was any kind of priority between task/script, so I was thinking of leveraging the `subject` field in the API response from Cirrus. However, the suggested change would certainly work too, as long as `task` takes precedence over `script` for the label

Reply:
Task has higher priority because in a build, a task is a set of scripts executed on the build machine